src/Core/Content/Product/SalesChannel/Review/ProductReviewSaveRoute.php

```diff
@@ -52,39 +51,28 @@ public function getDecorated(): AbstractProductReviewSaveRoute
     #[Route(path: '/store-api/product/{productId}/review', name: 'store-api.product-review.save', methods: ['POST'], defaults: ['_loginRequired' => true])]
     public function save(string $productId, RequestDataBag $data, SalesChannelContext $context): NoContentResponse
     {
-        $this->checkReviewsActive($context);
-
-        /** @var CustomerEntity $customer */
-        $customer = $context->getCustomer();
-
-        $languageId = $context->getContext()->getLanguageId();
-        $salesChannelId = $context->getSalesChannel()->getId();
-
-        $customerId = $customer->getId();
-
-        if (!$data->has('name')) {
-            $data->set('name', $customer->getFirstName());
+        $salesChannelId = $context->getSalesChannelId();
+        if ($this->config->getBool('core.listing.showReview', $salesChannelId) === false) {
+            throw new ReviewNotActiveExeption();
         }
 
-        if (!$data->has('lastName')) {
-            $data->set('lastName', $customer->getLastName());
-        }
+        $customer = $context->getCustomer();
 
-        if (!$data->has('email')) {
-            $data->set('email', $customer->getEmail());
```

Comment:
We dispatched ReviewFormEvent, which contains data such as name, last name, and email. However, you removed it from the data. I believe this change may cause a breaking change for those who subscribe to the ReviewFormEvent.